File: n3/3.rb
print "Введите количество чисел и желаемые числа: "
input = gets.split.map(&:to_i)

amount = input[0] # количество чисел
numbers = input[1..] # список чисел

odd = [] # список нечётных цифр
count = 0 # счётчик нечётных цифр

# перебор чисел
numbers.each do |num|
  num.to_s.each_char do |digit|
    if digit.to_i % 2 == 1 # проверка на нечётность
      count += 1
    end
  end
  odd << count # добавляем в список
  count = 0 # сброс счётчика
end

puts "Количество нечётных цифр в числах: #{odd.join(' ')}"
#

File: n3/3.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main() {
    int amount;
    cout << "Введите количество чисел и желаемые числа: ";
    cin >> amount;

    vector<string> numbers(amount);
    for (int i = 0; i < amount; i++) {
        cin >> numbers[i];
    }

    vector<int> odd; // список нечётных цифр
    int count = 0;   // счётчик нечётных цифр

    // перебор чисел
    for (string num : numbers) {
        for (char digit : num) {
            if ((digit - '0') % 2 == 1) { // проверка на нечётность
                count++;
            }
        }
        odd.push_back(count); // добавление в список
        count = 0; // сброс счётчика
    }

    cout << "Количество нечётных цифр в числах: ";
    for (int i : odd) {
        cout << i << " ";
    }
    cout << endl;
    return 0;
}
